ui/src/main/java/org/pentaho/di/ui/job/entries/job/JobEntryJobDialog.java

```diff
@@ -303,73 +303,51 @@ private void updateByReferenceField( RepositoryElementMetaInterface element ) {
     wByReference.setText( path );
   }
 
-  protected void pickFileVFS() {
-    FileDialog dialog = new FileDialog( shell, SWT.OPEN );
-    dialog.setFilterExtensions( Const.STRING_JOB_FILTER_EXT );
-    dialog.setFilterNames( Const.getJobFilterNames() );
+  private void pickFileVFS() {
     String prevName = jobMeta.environmentSubstitute( getPath() );
-    String parentFolder = null;
-    try {
-      parentFolder =
-        KettleVFS.getFilename( KettleVFS
-          .getFileObject( jobMeta.environmentSubstitute( jobMeta.getFilename() ) ).getParent() );
-    } catch ( Exception e ) {
-      // not that important
-    }
     if ( !Utils.isEmpty( prevName ) ) {
       try {
         if ( KettleVFS.fileExists( prevName ) ) {
-          dialog.setFilterPath( KettleVFS.getFilename( KettleVFS.getFileObject( prevName ).getParent() ) );
+          selectJob( ProviderFilterType.LOCAL );
+          return;
```

Comment:
Since line 311 is now the same as 330, could this if block be eliminated and have line 313 refactor as:
`if ( !KettleVFS.fileExists( prevName ) )`

Reply:
@cseverino789 changed.